ocrd_segment/classify_formdata_layout.py

```diff
@@ -722,86 +756,96 @@ def boundingbox(mask):
     cmax = np.max(indcs[1])
     return cmin, rmin, cmax - cmin, rmax - rmin
 
-@nb.njit(parallel=True, cache=True, nogil=True)
-def morphmasks(instances, components):
-    for i in nb.prange(len(instances)):
-        bbox, score, class_id, mask, category = instances[i]
-        # find closure in connected components
-        complabels = np.unique(mask * components)
-        left, top, w, h = boundingbox(mask)
-        right = left + w
-        bottom = top + h
-        if NP_POSTPROCESSING_OUTER:
-            # overwrite pixel mask from (padded) outer bbox
-            for label in complabels:
-                if not label:
-                    continue # bg/white
-                leftc, topc, wc, hc = boundingbox(components==label)
-                rightc = leftc + wc
-                bottomc = topc + hc
-                if wc > 2 * w or hc > 2 * h:
-                    continue # huge (non-text?) component
-                # intersection over component too small?
-                if (min(right, rightc) - max(left, leftc)) * \
-                   (min(bottom, bottomc) - max(top, topc)) < IOCC_THRESHOLD * wc * hc:
-                    continue # too little overlap
-                newleft = min(left, leftc)
-                newtop = min(top, topc)
-                newright = max(right, rightc)
-                newbottom = max(bottom, bottomc)
-                if (newright - newleft) > 2 * w or (newbottom - newtop) > 1.5 * h:
-                    continue # 
+#@nb.njit(parallel=True, cache=True, nogil=True)
+#@nb.njit(parallel=True)
+def morphmasks(args):
+    shared_masks_np = np.ctypeslib.as_array(shared_arr).reshape(shared_shape1)
+    shared_components_np = np.ctypeslib.as_array(second_shared_array).reshape(shared_shape2)
+    mask = shared_masks_np[args, : , :]
+    # find closure in connected shared_components_np
+    complabels = np.unique(mask * shared_components_np)
```

Comment:
Again, identifiers ...
```suggestion
def morphmasks(instance):
    masks = np.ctypeslib.as_array(shared_masks).reshape(shared_masks_shape)
    components = np.ctypeslib.as_array(shared_components).reshape(shared_components_shape)
    mask = masks[instance]
    # find closure in connected components
    complabels = np.unique(mask * components)
```